### PR TITLE
Intermittent failures in CloseTest and JDBCCustomTypesTest

### PR DESCRIPTION
**Pull request by tsegismont**
Two commits fixing the remaining (known) issues in the JDBC client test suite.

### CODE REVIEW

**Comment by tsegismont** [comment]
@pmlopes can you have a look? Thanks

File: src/test/java/io/vertx/ext/jdbc/CloseTest.java [hunk 1]
```diff
@@ -138,8 +140,16 @@ private void close(String deploymentId, boolean expectedDsThreadStatus) throws E
       closeLatch.countDown();
     }));
     awaitLatch(closeLatch);
-    assertTrue(!getConnThread.get(0).isAlive());
-    poolThreads.forEach(t -> assertEquals(t.isAlive(), expectedDsThreadStatus));
+    long start = System.nanoTime();
+    while (true) {
```
Comment (vietj) [inline src/test/java/io/vertx/ext/jdbc/CloseTest.java]: rather use milli seconds and use

```
while ((System.currentMillis - start < 5000) { ... }
```

more readable

File: src/test/java/io/vertx/ext/jdbc/JDBCCustomTypesTest.java [hunk 2]
```diff
@@ -49,18 +46,17 @@
...
-    }
-  }
+  private JDBCClient client;
 
   @Before
   public void setUp() throws Exception {
+    JsonObject config = ConfigFactory.createConfigForH2();
+    Connection conn = DriverManager.getConnection(config.getString("url"));
+    for (String sql : SQL) {
+      conn.createStatement().execute(sql);
```
Comment (vietj) [inline src/test/java/io/vertx/ext/jdbc/JDBCCustomTypesTest.java]: statement should be closed

File: src/test/java/io/vertx/ext/jdbc/ConfigFactory.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author Thomas Segismont
+ */
+public class ConfigFactory {
+
+  private static final AtomicInteger idGen = new AtomicInteger();
+
+  public static JsonObject createConfigForH2() {
```
Comment (vietj) [inline src/test/java/io/vertx/ext/jdbc/ConfigFactory.java]: can you just move this method in the base class and avoid to introduce a class only for that purpose ?

**Comment by tsegismont** [comment]
The intention is to use this class in new tests or when refactoring
existing tests.

2016-11-22 9:40 GMT+01:00 Julien Viet <notifications@github.com>:

**Comment by tsegismont** [comment]
Done

2016-11-22 9:37 GMT+01:00 Julien Viet <notifications@github.com>:

**Comment by tsegismont** [comment]
Done, as well as connection closing.

Note that there was a reason for this: by default, H2 closes the database
when the last connection is closed. So I had to change the JDBC url so that
the DB is no longer autoclosed, and change the test to manually shutdown
the DB.

2016-11-22 9:39 GMT+01:00 Julien Viet <notifications@github.com>:

**Pull request merged**